discontinuity_generic.py

```diff
@@ -0,0 +1,644 @@
+                    for form in forms
+                ]
+                for forms in self.J
+            ]
+
+            coeffs_a = [
+                [
+                    (
+                        {}
+                        if form is None
+                        else dolfinx.cpp.fem.pack_coefficients(form._cpp_object)
+                    )
+                    for form in forms
+                ]
+                for forms in self.J
+            ]
+
+            dolfinx.fem.petsc.assemble_vector_block(
+                self.b,
+                self.F,
+                self.J,
+                bcs=self.bcs,
+                x0=self.x,
+                alpha=-1.0,
+                coeffs_a=coeffs_a,
+                constants_a=constants_a,
+                coeffs_L=coeffs_L,
+                constants_L=constants_L,
+            )
+            self.b.ghostUpdate(
+                PETSc.InsertMode.INSERT_VALUES, PETSc.ScatterMode.FORWARD
+            )
+
+            # Assemble Jacobian
+            self.A.zeroEntries()
+            dolfinx.fem.petsc.assemble_matrix_block(
+                self.A, self.J, bcs=self.bcs, constants=constants_a, coeffs=coeffs_a
+            )
+            self.A.assemble()
+
+            self._solver.solve(self.b, self.dx)
+            # self._solver.view()
+            assert (
+                self._solver.getConvergedReason() > 0
+            ), "Linear solver did not converge"
+            offset_start = 0
+            for s in self.w:
+                num_sub_dofs = (
+                    s.function_space.dofmap.index_map.size_local
+                    * s.function_space.dofmap.index_map_bs
+                )
+                s.x.petsc_vec.array_w[:num_sub_dofs] -= (
+                    beta * self.dx.array_r[offset_start : offset_start + num_sub_dofs]
+                )
+                s.x.petsc_vec.ghostUpdate(
+                    addv=PETSc.InsertMode.INSERT, mode=PETSc.ScatterMode.FORWARD
+                )
+                offset_start += num_sub_dofs
+            # Compute norm of update
+
+            correction_norm = self.dx.norm(0)
+            print(f"Iteration {i}: Correction norm {correction_norm}")
+            if correction_norm < tol:
+                break
+            i += 1
+
+    def __del__(self):
+        self.A.destroy()
+        self.b.destroy()
+        self.dx.destroy()
+        self._solver.destroy()
+        self.x.destroy()
+
+
+def transfer_meshtags_to_submesh(
+    mesh, entity_tag, submesh, sub_vertex_to_parent, sub_cell_to_parent
+):
+    """
+    Transfer a meshtag from a parent mesh to a sub-mesh.
+    """
+
+    tdim = mesh.topology.dim
+    cell_imap = mesh.topology.index_map(tdim)
+    num_cells = cell_imap.size_local + cell_imap.num_ghosts
+    mesh_to_submesh = np.full(num_cells, -1)
+    mesh_to_submesh[sub_cell_to_parent] = np.arange(
+        len(sub_cell_to_parent), dtype=np.int32
+    )
+    sub_vertex_to_parent = np.asarray(sub_vertex_to_parent)
```

Comment:
```suggestion
```
I don't think we need this anymore after: https://github.com/FEniCS/dolfinx/pull/3112/

Reply:
Done